src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -597,14 +597,15 @@ private void ValidateUnsafeParameters(DiagnosticBag diagnostics, ImmutableArray<
         private BoundLambda ReallyInferReturnType(NamedTypeSymbol delegateType, ImmutableArray<TypeWithAnnotations> parameterTypes, ImmutableArray<RefKind> parameterRefKinds)
         {
             var diagnostics = DiagnosticBag.GetInstance();
+            // TypeWithAnnotations
             var lambdaSymbol = new LambdaSymbol(
                 Binder.Compilation,
                 Binder.ContainingMemberOrLambda,
                 _unboundLambda,
                 parameterTypes,
                 parameterRefKinds,
                 refKind: CodeAnalysis.RefKind.None,
-                returnType: default,
+                returnType: delegateType is null ? TypeWithAnnotations.Create(new ExtendedErrorTypeSymbol(this.Binder.Compilation, "", arity: 0, errorInfo: null, unreported: false)) : default,
```

Comment:
> returnType: delegateType is null ? TypeWithAnnotations.Create(new ExtendedErrorTypeSymbol(this.Binder.Compilation, "", arity: 0, errorInfo: null, unreported: false)) : default, [](start = 15, length = 177)

It is not clear what is the benefit of this change. Could you please elaborate?

Reply:
`delegateType` is null only in the error scenario when it is being called from `ReallyBindForErrorRecovery`.  Since it will be returned into the bound tree, we need to ensure that the returned expressions have been converted.  So we produce an error type as the conversion target.  However, I am going to try another approach, as this undermines the selection of an appropriate return type for IDE scenarios.

---
In reply to: [328268618](https://github.com/dotnet/roslyn/pull/38750#discussion_r328268618) [](ancestors = 328268618)